Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -597,6 +612,15 @@ public struct PackageBuilder {
             products.append(product)
         }
 
+        // Create a product for the entire package.
+        // FIXME: This is to be done only when parsing the manifest in Swift 3 mode.
+        if createImplicitProduct {
```

Comment:
Should we also do this only if the package doesn't declare any explicit products?